### ISSUE TITLE
Handle Partially Synced Nodes

### ISSUE THREAD

**Issue by nisdas** [prysmaticlabs/prysm#555]
When a node runs the initial sync service, it first checks for stored state and if there is stored state it defaults to a regular sync. A PR would handle nodes who are only partially synced , and are not synced with the latest head of the current chain. 

This would involve modifying this portion of the initial-sync service:
```golang
 if stored {
		log.Info("Chain state detected, exiting initial sync")
		return
	}  
```

**Comment by prestonvanloon** [comment]
Blocking #622

**Comment by nisdas** [comment]
@nisdas Ah , no . This will not be. But it would be resolved when #743 is resolved
